clang/lib/Sema/Sema.cpp

```diff
@@ -1727,6 +1727,11 @@ class DeferredDiagnosticsEmitter
               << Sema::KernelConstStaticVariable;
           return;
         }
+        if (!VD->hasInit() &&
```

Comment:
What is the hasInit check for?

Reply:
It is needed to differentiate between a declaration of device_global variable and definition of it. The intention is to diagnose when externally defined device_global not marked with SYCL_EXTERNAL macro is referenced from SYCL device code. We have the similar diagnostic for SYCL_EXTERNAL and external functions.